packages/analyzer/src/features/analyse-phase/creators/createClass.js

```diff
@@ -207,43 +208,57 @@ export function getDefaultValuesFromConstructorVisitor(source, classTemplate, co
          */
         node.body?.statements?.filter((statement) => statement.kind === ts.SyntaxKind.ExpressionStatement)
           .filter((statement) => statement.expression.kind === ts.SyntaxKind.BinaryExpression)
-          .forEach((statement) => {
-            let existingMember = classTemplate?.members?.find(member => statement.expression?.left?.name?.getText() === member.name && member.kind === 'field');
+          .forEach((statement) => mapBinaryExpression(source, classTemplate, context, statement, statement.expression));
```

Comment:
As seen in the [AST](https://ts-ast-viewer.com/#code/MYGwhgzhAEAu0FMAesEDsAmMASAVAsgDICiICAtuvAN7TAD2aEsATgK7Cz0sAUAlNFqwAFgEsIAOlTMADi3oyAjNAC80AIQAGADRwxk6bDkKATKugAiAGIBVWxegBfJ3TbN65UhSqSMCAGaiaAg8Fv709AC0CCAWurB8ANwAUMlAA), multiple expression can belong to a single statement.

![image](https://user-images.githubusercontent.com/16068444/126983539-174958aa-f268-47b8-8796-b7e4ee1e4cde.png)


This case happens when they are comma separated, which is the main subject of this PR.

So by default we want to pass the default expression of the statement, but inside the file, [here](https://github.com/open-wc/custom-elements-manifest/blob/1f8f907fa5543c648ed01109e4a5803900d4aea4/packages/analyzer/src/features/analyse-phase/creators/createClass.js#L226), we pass the same statement, since they both are part of the same statement, therefore part of same jsdoc, etc. nodes, BUT INSTEAD we pass the actual expressions of the values assignments.

e.g.
```javascript
constructor() {
   /** These are values which are part of the same statement */
     this.testprop1 = !0, // expression.left
      this.testprop2 = "FUFU"; // expression.right
}
```

Reply:
Currently we pass both `statement` and `statement.expression` to the function

But `expression` is a property of `statement`, which we already pass. So inside `mapBinaryExpression` we should either destructure `expression` from `statement` or access it like `statement.expression`. There's no need to pass both to the function, because the former contains the latter.

Also `source` is not needed, because you can call `getSourceFile` on any AST node

Reply:
I'm not sure if we're on the same page here or if I'm missing something 😀

If we look for example a case like 

```javascript
        constructor() { 
            this.testprop1 = !0,
            this.testprop2 = "FUFU",
            this.testprop3 = true,
            this.testprop4 = false 
        }
```

The AST tree would look like this:

![image](https://user-images.githubusercontent.com/16068444/127032881-f797878a-9888-47f3-8633-1d76f0bad4d1.png)

[Link to ast](https://ts-ast-viewer.com/#code/MYGwhgzhAEAu0FMAesEDsAmMASAVAsgDICiICAtuvAN7QBQ0j0wA9mhLAE4CuwsLnABQBKaLQZNJsABYBLCADpUHAA6cWKgIzQAvNACEABgA0EyYxnylCVepUAmXdABEAMQCqH56fNNLi5Vg1DQBmJy5uBB9fODkAmyC7ABYnADMwEAgEehjGM3MAX2gi4G4OFnJSCipFDARU2TQEQWdUlhYAWgQQbzhhAG46OiA)


And looking at it, it's a clear case that we would need to recursively handle this situation, going down the tree 'till we get the actual expressions instead of just a comma separarated value expression.

And for this kind of case, the `statement` always keeps the same, since for the AST, all of the code from  `this.testprop1` till ` this.testprop4 = false` is under the same `statement`.

But the actual expression changes, as we go down the tree. This is why we would need to separate the expression from the statement since it might not be the direct descendant of the statement.

e.g. 

`this.testprop1` would be `expression.left.left.left` on the first pass through.